pkg/log/log_test.go

```diff
@@ -0,0 +1,203 @@
+	if err != nil {
+		return err
+	}
+
+	for _, file := range files {
+		sourceFilePath := path.Join(sourceDir, file.Name())
+		destFilePath := path.Join(destDir, file.Name())
+
+		if file.IsDir() {
+			if err = copyFiles(sourceFilePath, destFilePath); err != nil {
+				return err
+			}
+		} else {
+			sourceFile, err := os.Open(sourceFilePath)
+			if err != nil {
+				return err
+			}
+			defer sourceFile.Close()
+
+			destFile, err := os.Create(destFilePath)
+			if err != nil {
+				return err
+			}
+			defer destFile.Close()
+
+			if _, err = io.Copy(destFile, sourceFile); err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func TestAuditLogMessages(t *testing.T) {
+	Convey("Make a new controller", t, func() {
+		config := api.NewConfig()
+
+		config.Log = &api.LogConfig{Level: "debug", Output: "./zot.log", Audit: "./zot-audit.log"}
+
+		config.HTTP.Port = SecurePort1
+
+		htpasswdPath := makeHtpasswdFile()
+		defer os.Remove(htpasswdPath)
+		config.HTTP.Auth = &api.AuthConfig{
+			HTPasswd: api.AuthHTPasswd{
+				Path: htpasswdPath,
+			},
+		}
+
+		c := api.NewController(config)
+		dir, err := ioutil.TempDir("", "oci-repo-test")
+		if err != nil {
+			panic(err)
+		}
+		defer os.RemoveAll(dir)
+		err = copyFiles("../../test/data", dir)
+		if err != nil {
+			panic(err)
+		}
+		c.Config.Storage.RootDirectory = dir
+		go func() {
+			// this blocks
+			if err := c.Run(); err != nil {
+				return
+			}
+		}()
+
+		// wait till ready
+		for {
+			_, err := resty.R().Get(BaseURL1)
+			if err == nil {
+				break
+			}
+			time.Sleep(100 * time.Millisecond)
+		}
+
+		defer func() {
+			ctx := context.Background()
+			_ = c.Server.Shutdown(ctx)
+		}()
+
+		Convey("Open auditLog file", func() {
+			jsonFile, err := os.Open("./zot-audit.log")
+			if err != nil {
+				t.Log("Cannot open file")
+				panic(err)
+			}
+			defer jsonFile.Close()
+
+			Convey("Test GET request", func() {
+				resp, err := resty.R().SetBasicAuth(username, passphrase).
+					Get(BaseURL1 + "/v2/")
+				So(err, ShouldBeNil)
+				So(resp, ShouldNotBeNil)
+				So(resp.StatusCode(), ShouldEqual, 200)
+
+				byteValue, _ := ioutil.ReadAll(jsonFile)
+				So(len(byteValue), ShouldEqual, 0)
+			})
+
+			Convey("Test POST request", func() {
+				path := "/v2/" + AuthorizedNamespace + "/blobs/uploads/"
+				resp, err := resty.R().SetBasicAuth(username, passphrase).
+					Post(BaseURL1 + path)
+				So(err, ShouldBeNil)
+				So(resp, ShouldNotBeNil)
+				So(resp.StatusCode(), ShouldEqual, 202)
+
+				byteValue, _ := ioutil.ReadAll(jsonFile)
+				So(len(byteValue), ShouldNotEqual, 0)
+
+				var auditLog AuditLog
+				err = json.Unmarshal(byteValue, &auditLog)
+				if err != nil {
+					panic(err)
+				}
+
+				So(auditLog.Subject, ShouldEqual, username)
+				So(auditLog.Action, ShouldEqual, http.MethodPost)
+				So(auditLog.Status, ShouldEqual, http.StatusAccepted)
+				So(auditLog.Object, ShouldEqual, path)
+			})
+		})
+
+		err = os.Remove("./zot.log")
```

Comment:
Why not call os.Remove using defer,  similar to https://github.com/anuvu/zot/pull/199/files#diff-c15ccbf2169a85a01173cece538ff8583ee6905d27c8ffbdedc4ec2d953bb50eR111. That would make sure they are deleted.